### ISSUE TITLE
Output divergence in MLP with torch.compile + AMP when biases are enabled

### ISSUE THREAD

**Issue by ChenNingCong** [meta-pytorch/tritonbench#607]
The model's output diverges significantly when using torch.compile with Automatic Mixed Precision (AMP) (float32-bfloat16) on a simple MLP layer that includes bias terms in its linear layers. The output is identical to the uncompiled model when the biases are disabled. This issue occurs within a basic MLP structure: Linear Layer -> ReLU -> Linear Layer.

The minimal reproducible example is located on the main branch: https://github.com/ChenNingCong/tritonbench

Steps to Reproduce and Results
1. Running the model with biases (Divergent results)
```
python run.py --op Mlp --metrics accuracy --use_bias
```
Output:
```
Exception during accuracy check: Tensor-likes are not close!

Mismatched elements: 55273 / 491520 (11.2%)
Greatest absolute difference: 0.003173828125 at index (3, 0, 3, 178) (up to 1e-05 allowed)
Greatest relative difference: 5440.0 at index (6, 2, 0, 136) (up to 0.016 allowed)
```
2. Running the model without biases (Matching results)
When bias terms are explicitly disabled in the linear layers, the compiled and uncompiled models yield exact matches.

```
python run.py --op Mlp --metrics accuracy --use_bias
```
Output (all the results are matched exactly):
```
  x_val    compile_mlp-accuracy
-------  ----------------------
      0                       1
      1                       1
      2                       1
      3                       1
      4                       1
      5                       1
      6                       1
      7                       1
      8                       1
      9                       1
```
This is tested on A100, L40S and H100. It seems that at least on H100, triton fuses bias with ReLU when torch.compile is enabled. Otherwise bias is fused into matmul cuda kernel.

**Comment by xuzhao9** [comment]
I can reproduce this on Blackwell too.

**Comment by xuzhao9** [comment]
Took some time to look into this. The reason is not because of Triton, instead it is because nvidia's addmm kernel is losing accuracy comparing to torch.mm+torch.add.

Without compile (eager mode), the module is decomposed to 1. torch.addmm and 2. torch.clamp_min

With compile, the module is decomposed to 1. torch.mm 2. triton_poi_fused_addmm_relu_view_0

If we take "torch.mm+torch.add+torch.clamp_min" as the gold standard of the numerics, we see with compile matches the numerics, but eager mode does not match.

Proof:
With https://github.com/meta-pytorch/tritonbench/pull/609:

```
$ python run.py --op mlp --metrics accuracy --baseline aten2 --only aten,triton,aten2
WARNING:tritonbench.utils.triton_op:First-k mode: Selected 1 sequential inputs starting from index 0 (total available: 1)
WARNING:tritonbench.utils.triton_op:Input IDs to run: [0]
  0%|                                                                                                                                                                                                | 0/1 [00:00<?, ?it/s]WARNING:tritonbench.utils.triton_op:Running input ID 0:
  x_val
-------
      0
INFO:tritonbench.utils.triton_op:Took 37.85ms to get benchmark function for aten2
INFO:tritonbench.utils.triton_op:Took 36.77ms to get benchmark function for aten
WARNING:tritonbench.utils.triton_op:Exception during accuracy check: Tensor-likes are not close!

Mismatched elements: 2470 / 1966080 (0.1%)
Greatest absolute difference: 0.0078125 at index (0, 1263) (up to 1e-05 allowed)
Greatest relative difference: inf at index (5, 97) (up to 0.016 allowed)
INFO:tritonbench.utils.triton_op:Took 0.01ms to get benchmark function for triton
WARNING:tritonbench.utils.triton_op:Completed input ID 0:

  x_val    aten-accuracy    triton-accuracy
-------  ---------------  -----------------
      0                0                  1
average                0                  1
```

`aten`: torch.addmm + torch.clamp_min
`aten2`: torch.mm + torch.add + torch.clamp_min
`triton`: torch.mm + triton_poi_fused_addmm_relu_view_0

It shows when using `aten2` as the baseline, triton matches but eager mode does not.

**Comment by xuzhao9** [comment]
Reopening as this is a PyTorch compiler decomposition issue.

**Comment by xuzhao9** [comment]
Should be fixed by https://github.com/pytorch/pytorch/pull/176848
